frog/imports/client/Wiki/index.js

```diff
@@ -182,12 +194,76 @@ class WikiComp extends React.Component<WikiCompPropsT, WikiCompStateT> {
       );
     }
 
+    // Show locked modal if the wiki is locked
+    if (
+      this.wikiDoc.data.settings?.locked &&
+      !this.wikiDoc.data.owners.find(x => x === Meteor.userId())
+    ) {
+      this.setState({ currentPageObj: null });
+      this.props.showModal(<LockedModal />);
+      return;
+    } else {
+      this.initialLoad = true;
+      this.props.hideModal();
+    }
+    // Ask for password if wiki access is password restricted
+    if (
+      this.wikiDoc.data.settings?.restrict === 'view' &&
+      !this.wikiDoc.data.users?.find(x => x === Meteor.userId()) &&
+      !this.wikiDoc.data.owners.find(x => x === Meteor.userId())
+    ) {
+      this.setState({ currentPageObj: null });
+      const passwordPromise = new Promise(resolve => {
+        this.props.showModal(
+          <PasswordModal
+            callback={resolve}
+            hideModal={this.props.hideModal}
+            actualPassword={this.wikiDoc.data.settings?.password}
+          />
+        );
+      });
+      const result = await passwordPromise;
+      if (!result) {
+        this.props.showModal(
+          <Modal
+            title="Unable to access Wiki"
+            actions={[
+              {
+                title: 'OK',
+                callback: () => {
+                  this.props.hideModal();
+                  this.loadWikiDoc();
+                }
+              }
+            ]}
+          >
+            This wiki has been password protected.
+          </Modal>
+        );
+      } else addUser(this.wikiDoc, Meteor.userId());
+      return;
+    } else {
+      if (
+        this.wikiDoc.data.users?.find(x => x === Meteor.userId()) === undefined
+      )
+        addUser(this.wikiDoc, Meteor.userId());
+      this.initialLoad = true;
+      this.props.hideModal();
+    }
     wikiStore.setPages(this.wikiDoc.data.pages);
     this.setState({
       pagesData: wikiStore.pages,
-      settings: this.wikiDoc.data.settings
+      settings: this.wikiDoc.data.settings,
+      isOwner:
+        this.wikiDoc.data.owners.find(x => x === Meteor.userId()) !== undefined
     });
-
+    if (
+      (this.wikiDoc.data.settings?.readOnly ||
+        this.wikiDoc.data.settings.restrict === 'edit') &&
```

Comment:
The allowed restrict values should be moved to constants to help with code maintenance